ip_read: BUG: rc % (((((((ip->data.sf))) & 0x00000038 ) >> 0) >> 3) * ((((ip->data.sf)) & 0xff000000 ) >> 24)) != 0

<!-- Please fill the following if you have found an issue, delete if otherwise. -->
Hi there! Seems like i got myself a little bug.  In a track "Corrupt system"(By VHS Glitch, album is Moral Decay), i'm getting this after some time of playing and cmus crashes

```
ip_read: BUG: rc % (((((((ip->data.sf))) & 0x00000038 ) >> 0) >> 3) * ((((ip->data.sf)) & 0xff000000 ) >> 24)) != 0
```
It seems that some of the tracks in this album is causing this bug too. The other ones is playing absolutely great. I'm actually listening CD-rip in flac with ~1500 kbps, so it's not caused by some badly converted song.

1) `uname -a`:
```
Linux laptop.scvh 4.9.5-200.fc25.x86_64 #1 SMP Fri Jan 20 12:24:16 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

2) `cmus --version`:
```
cmus v2.7.1
```

3) `cmus --plugins`:
```
Input Plugins: /usr/lib64/cmus/ip
  ffmpeg:
    Priority: 300
    File Types: aa aac ac3 aif aifc aiff ape au fla flac m4a m4b mka mkv mp+ mp2 mp3 mp4 mpc mpp ogg shn tak tta wav webm wma wv
    MIME Types:
  mad:
    Priority: 55
    File Types: mp3 mp2
    MIME Types: audio/mpeg audio/x-mp3 audio/x-mpeg
  wavpack:
    Priority: 50
    File Types: wv
    MIME Types: audio/x-wavpack
  cue:
    Priority: 50
    File Types:
    MIME Types: application/x-cue
  aac:
    Priority: 50
    File Types: aac
    MIME Types: audio/aac audio/aacp
  mp4:
    Priority: 50
    File Types: mp4 m4a m4b
    MIME Types:
  wav:
    Priority: 50
    File Types: wav
    MIME Types:
  vorbis:
    Priority: 50
    File Types: ogg oga ogx
    MIME Types: application/ogg audio/x-ogg
  opus:
    Priority: 50
    File Types: opus
    MIME Types:
  flac:
    Priority: 50
    File Types: flac fla
    MIME Types:
  modplug:
    Priority: 50
    File Types: mod s3m xm it 669 amf ams dbm dmf dsm far mdl med mtm okt ptm stm ult umx mt2 psm
    MIME Types:
  mpc:
    Priority: 50
    File Types: mpc mpp mp+
    MIME Types: audio/x-musepack

Output Plugins: /usr/lib64/cmus/op
  pulse
  alsa
  ao
```

That's weird, I cannot build cmus from this branch https://github.com/nefthy/cmus/tree/issue_644
After `make` linker complains:
```
   LD     cmus
ape.o: In function `ape_parse_one':
ape.c:(.text+0x4ef): undefined reference to `xstrndup'
browser.o: In function `browser_up':
browser.c:(.text+0xd2f): undefined reference to `xstrndup'
collect2: error: ld returned 1 exit status
Makefile:52: recipe for target 'cmus' failed
make: *** [cmus] Error 1
```

--- a/ip/flac.c
+++ b/ip/flac.c
@@ -217,8 +217,6 @@ static void metadata_cb(const Dec *dec, const FLAC__StreamMetadata *metadata, vo
 				break;
 			case 20:
 			case 24:
-				bits = 24;
-				break;
 			case 32:
 				bits = 32;
 				break;

